mipengine/controller/algorithm_executor_node_data_objects.py

```diff
@@ -0,0 +1,355 @@
+from abc import ABC
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Union
+
+from mipengine.node_tasks_DTOs import NodeLiteralDTO
+from mipengine.node_tasks_DTOs import NodeSMPCDTO
+from mipengine.node_tasks_DTOs import NodeSMPCValueDTO
+from mipengine.node_tasks_DTOs import NodeTableDTO
+from mipengine.node_tasks_DTOs import NodeUDFDTO
+from mipengine.node_tasks_DTOs import TableSchema
+from mipengine.node_tasks_DTOs import UDFKeyArguments
+from mipengine.node_tasks_DTOs import UDFPosArguments
+
+
+class NodeData(ABC):
+    """
+    NodeData are located into one specific Node.
+    """
+
+    pass
+
+
+class NodeTable(NodeData):
+    def __init__(self, table_name):
+        self._full_name = table_name
+        full_name_split = self._full_name.split("_")
+        self._table_type = full_name_split[0]
+        self._node_id = full_name_split[1]
+        self._context_id = full_name_split[2]
+        self._command_id = full_name_split[3]
+        self._command_subid = full_name_split[4]
+
+    @property
+    def full_table_name(self) -> str:
+        return self._full_name
+
+    @property
+    def table_type(self) -> str:
+        return self._table_type
+
+    @property
+    def node_id(self) -> str:
+        return self._node_id
+
+    @property
+    def context_id(self) -> str:
+        return self._context_id
+
+    @property
+    def command_id(self) -> str:
+        return self._command_id
+
+    @property
+    def command_subid(self) -> str:
+        return self._command_subid
+
+    def without_node_id(self) -> str:
+        return (
+            self._table_type
+            + "_"
+            + self._context_id
+            + "_"
+            + self._command_id
+            + "_"
+            + self._command_subid
+        )
+
+    def __repr__(self):
+        return self.full_table_name
+
+
+class NodeSMPCTables(NodeData):
+    template: NodeTable
+    add_op: NodeTable
+    min_op: NodeTable
+    max_op: NodeTable
+    union_op: NodeTable
+
+    def __init__(self, template, add_op, min_op, max_op, union_op):
+        self.template = template
+        self.add_op = add_op
+        self.min_op = min_op
+        self.max_op = max_op
+        self.union_op = union_op
```

Comment:
use pydantic?

Reply:
`NodeSMPCTables` and `NodeData` are not DTOs and could have properties/methods. So I prefer for it not to be a BaseModel.